extensions/git/src/git.ts

```diff
@@ -1730,6 +1730,64 @@ export class Repository {
 		await this.exec(args);
 	}
 
+	async fetchPruneAndDelete(): Promise<void> {
+		const fetchPruneOutput = await this.exec(['fetch', '--prune']);
+		const { exitCode, stderr } = fetchPruneOutput;
+
+		if (exitCode) {
+			throw new GitError({
+				message: 'Could not fetch.',
+				exitCode
+			});
+		}
+
+		// The outputs of the fetch prune is at stderr
+		if (!stderr) {
+			throw new GitError({
+				message: 'Could not find pruned branches to delete.',
+			});
+		}
+
+		const branchRegex = /->\s*(\S+)/;
+		// [
+		// 	'From github.com:username/some-repo',
+		// 	' - [deleted]         (none)     -> origin/foo',
+		// 	' - [deleted]         (none)     -> origin/bar',
+		// 	''
+		// ]
+		const branches: string[] = stderr.split('\n')
+			// [
+			// 	'origin/foo',
+			// 	'origin/bar',
+			// ]
+			// TODO: origin is hardcoded
+			.map((el) => el.match(branchRegex)?.[1]?.replace('origin/', '') ?? '')
```

Comment:
```suggestion
			.map((el) => el.match(branchRegex)?.[1]?.replace('origin/', '') ?? '')
```

I don't think there will be edge cases for remote shorthand `origin`